api.lua

```diff
@@ -104,10 +108,26 @@ end
 
 -- color
 function changeToolColor(color)
-	local result = app.changeToolColor({ ["color"] = color, ["selection"] = true })
-
-	-- change text tool color too, because (with the default template) the TEX tool also uses that color
-	app.changeToolColor({ ["color"] = color, ["tool"] = "TEXT" })
+	-- get active tool
+	local activeToolInfo = getToolInfo("active")
+	if not activeToolInfo then return end
+	local toolType = activeToolInfo["type"] or "unknown"
+	-- get the info for that specific tool
+	local toolInfo = getToolInfo(toolType)
+	if not toolInfo then return end
+	-- Check if this tool really has a color property
+	if toolInfo["color"] ~= nil then
+		local ok = pcall(function()
+			app.changeToolColor({ ["color"] = color, ["selection"] = true })
+		end)
+		if ok and toolType ~= "text" then
```

Comment:
Why does it need to be `ok` in order to change the color of the text tool? If I change the color on the eraser I want the next LaTeX insert to be the new color.